### PR TITLE
ocamldep: do not process files during cli parsing.

### PR DESCRIPTION
**Pull request by dbuenzli**
See #9299.

### CODE REVIEW

File: driver/makedepend.ml [hunk 1]
```diff
@@ -562,6 +562,21 @@ let parse_map fname =
...
+  | Map of Misc.filepath (* -map option *)
+  | Src of Misc.filepath * file_kind option (* -impl, -intf or anon arg *)
+
+let process_dep_arg = function
+  | Map file -> parse_map file
+  | Src (file, file_kind) ->
+      match file_kind with
+      | None -> file_dependencies file
+      | Some MLI -> file_dependencies_as MLI file
+      | Some ML -> file_dependencies_as ML file
```
Comment (nojb) [inline driver/makedepend.ml]: I would shorten this as:
```
let process_dep_arg = function
  | Map file -> parse_map file
  | Src (file, None) -> file_dependencies file
  | Src (file, Some kind) -> file_dependencies_as kind file
```

**Review by nojb — APPROVED** [review]
LGTM. I left a style suggestion; up to you to take it or leave it.

File: driver/makedepend.ml (hunk 1 above)
Comment (dbuenzli) [reply driver/makedepend.ml]: Indeed ! That was a bit verbose :-)

**Comment by nojb** [comment]
Merged, thanks!

**Pull request merged**